File: Introduction-to-algorithms-3-edition/chapter2/Insertion-sort/Go/1/main.go
package main

import "fmt"

func main() {
	fmt.Println(insertionSort([]int{5, 2, 4, 6, 1, 3}))
}

func insertionSort(arr []int) []int {
	j := 1
	for j < len(arr) {
		key := arr[j]
		i := j - 1
		for i >= 0 && arr[i] > key {
			arr[i + 1] = arr[i]
			i--
		}
		arr[i + 1] = key
		j++
	}
	return arr
}

File: Introduction-to-algorithms-3-edition/chapter2/Insertion-sort/Ruby/1/main.rb

# @param {Array} arr
# @return {Array}
def insertion_sort(arr)
    j = 1
    for j in j...arr.length do
        key = arr[j]
        i = j - 1
        while i > 0 and arr[i] > key
            arr[i + 1] = arr[i]
            i--
        end
        arr[i + 1] = key
    end
end
